### PR TITLE
Code cleanup around the Amqp device files

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs [hunk 1]
```diff
@@ -7,60 +7,61 @@
 using System.Threading.Tasks;
 using Microsoft.Azure.Devices.Shared;
 using Microsoft.Azure.Devices.Client.Exceptions;
-using Microsoft.Azure.Devices.Client.Transport.AmqpIoT;
+using Microsoft.Azure.Devices.Client.Transport.AmqpIot;
```
Comment (drwill-ms) [inline iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs:10]: Changed the casing of the namespace to Iot from IoT

File: iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs [hunk 2]
```diff
@@ -7,60 +7,61 @@
...
-                Logging.Associate(this, amqpCbsLink, $"{nameof(_amqpIoTCbsLink)}");
+                Logging.Associate(this, amqpCbsLink, nameof(_amqpIoTCbsLink));
             }
         }
 
         public async Task InitLoopAsync(TimeSpan timeout)
         {
             if (Logging.IsEnabled)
             {
-                Logging.Enter(this, timeout, $"{nameof(InitLoopAsync)}");
```
Comment (drwill-ms) [inline iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs:49]: Why wrap something that is a string inside an interpolated string?

File: iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs [hunk 3]
```diff
@@ -7,60 +7,61 @@
...
+                Logging.Enter(this, timeout, nameof(InitLoopAsync));
             }
 
             CancellationTokenSource oldTokenSource = _cancellationTokenSource;
             _cancellationTokenSource = new CancellationTokenSource();
             CancellationToken newToken = _cancellationTokenSource.Token;
             oldTokenSource?.Cancel();
+            oldTokenSource?.Dispose();
```
Comment (drwill-ms) [inline iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs:56]: Looks like we were missing dispose for this

File: iothub/device/src/Transport/Amqp/AmqpConnectionHolder.cs [hunk 4]
```diff
@@ -144,49 +144,51 @@ public async Task<IAmqpAuthenticationRefresher> CreateRefresherAsync(DeviceIdent
         {
             if (Logging.IsEnabled)
             {
-                Logging.Enter(this, deviceIdentity, timeout, $"{nameof(CreateRefresherAsync)}");
+                Logging.Enter(this, deviceIdentity, timeout, nameof(CreateRefresherAsync));
             }
 
-            AmqpIoTConnection amqpIoTConnection = await EnsureConnectionAsync(timeout).ConfigureAwait(false);
-            IAmqpAuthenticationRefresher amqpAuthenticator = await amqpIoTConnection.CreateRefresherAsync(deviceIdentity, timeout).ConfigureAwait(false);
+            AmqpIotConnection amqpIoTConnection = await EnsureConnectionAsync(timeout).ConfigureAwait(false);
```
Comment (abhipsaMisra) [inline iothub/device/src/Transport/Amqp/AmqpConnectionHolder.cs:150]: amqpIoTConnection (variable name)?

File: iothub/device/src/Transport/Amqp/AmqpUnitManager.cs [hunk 5]
```diff
@@ -14,8 +14,8 @@ internal class AmqpUnitManager : IAmqpUnitManager
     {
         private static readonly AmqpUnitManager s_instance = new AmqpUnitManager();
 
-        private IDictionary<string, IAmqpUnitManager> _amqpConnectionPools;
-        private readonly object _lock = new object();
+        private readonly IDictionary<string, IAmqpUnitManager> _amqpConnectionPools;
+        private readonly object _connectionPoolLock = new object();
```
Comment (drwill-ms) [inline iothub/device/src/Transport/Amqp/AmqpUnitManager.cs:18]: A more descriptive name

File: iothub/device/src/Transport/AmqpIot/AmqpIotCbsLink.cs [hunk 6]
```diff
@@ -7,31 +7,39 @@
 using Microsoft.Azure.Devices.Shared;
 using Microsoft.Azure.Devices.Client.Extensions;
 
-namespace Microsoft.Azure.Devices.Client.Transport.AmqpIoT
+namespace Microsoft.Azure.Devices.Client.Transport.AmqpIot
 {
-    internal class AmqpIoTCbsLink
+    internal class AmqpIotCbsLink
```
Comment (drwill-ms) [inline iothub/device/src/Transport/AmqpIot/AmqpIotCbsLink.cs:12]: Class names changed from IoT to Iot too

File: iothub/device/src/Transport/AmqpIot/AmqpIotLinkFactory.cs [hunk 7]
```diff
@@ -6,48 +6,43 @@
...
-            {
-                return new ReceivingAmqpLink(session, settings);
-            }
-            else
-            {
-                return new SendingAmqpLink(session, settings);
-            }
+            return settings.IsReceiver()
+                ? new ReceivingAmqpLink(session, settings)
+                : (AmqpLink)new SendingAmqpLink(session, settings);
```
Comment (abhipsaMisra) [inline iothub/device/src/Transport/AmqpIot/AmqpIotLinkFactory.cs:40]: q - why did only one of the code paths require an explicit cast?

File: iothub/device/src/Transport/AmqpIot/AmqpIotMessageConverter.cs [hunk 8]
```diff
@@ -96,9 +95,9 @@ public static void UpdateMessageHeaderAndProperties(AmqpMessage amqpMessage, Mes
                     message.ContentEncoding = amqpMessage.Properties.ContentEncoding.Value;
                 }
 
-                message.UserId = amqpMessage.Properties.UserId.Array != null
-                    ? Encoding.UTF8.GetString(amqpMessage.Properties.UserId.Array, 0 /*index*/, amqpMessage.Properties.UserId.Array.Length)
-                    : null;
+                message.UserId = amqpMessage.Properties.UserId.Array == null
```
Comment (drwill-ms) [inline iothub/device/src/Transport/AmqpIot/AmqpIotMessageConverter.cs:98]: Positive equalities are easier to read

File: iothub/device/src/Transport/AmqpIot/AmqpIotMessageConverter.cs [hunk 9]
```diff
@@ -305,12 +305,10 @@ public static MethodRequestInternal ConstructMethodRequestFromAmqpMessage(AmqpMe
                 methodRequestId = amqpMessage.Properties.CorrelationId?.ToString();
             }
 
-            if ((sections & SectionFlag.ApplicationProperties) != 0)
```
Comment (drwill-ms) [inline iothub/device/src/Transport/AmqpIot/AmqpIotMessageConverter.cs:308]: nested if with only 1 code path can be a compound if statement

File: iothub/device/src/Transport/AmqpIot/AmqpIotMessageConverter.cs [hunk 10]
```diff
@@ -365,13 +363,13 @@ private static bool TryGetNetObjectFromAmqpObject(object amqpObject, MappingType
                     break;
 
                 case PropertyValueType.Unknown:
-                    if (amqpObject is AmqpSymbol)
+                    if (amqpObject is AmqpSymbol amqpSymbol)
```
Comment (drwill-ms) [inline iothub/device/src/Transport/AmqpIot/AmqpIotMessageConverter.cs:366]: Make use of type check with cast

File: iothub/device/src/Transport/AmqpIoT/AmqpIoTOutcome.cs [hunk 11]
```diff
@@ -1,50 +0,0 @@
-﻿// Copyright (c) Microsoft. All rights reserved.
```
Comment (drwill-ms) [inline iothub/device/src/Transport/AmqpIoT/AmqpIoTOutcome.cs:1]: Weird diff for this file and next. They appear below, but git didn't pick up it was just a rename.

File: iothub/device/src/Transport/AmqpIot/AmqpIotResultAdapter.cs [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+        Rejected
+    }
+
+    internal static class AmqpIotResultAdapter
+    {
+        internal static Outcome GetResult(AmqpIotDisposeActions amqpIotConstants)
+        {
+            return amqpIotConstants switch
+            {
+                AmqpIotDisposeActions.Accepted => new Accepted(),
```
Comment (drwill-ms) [inline iothub/device/src/Transport/AmqpIot/AmqpIotResultAdapter.cs:22]: Besides rename, main code change was using the new `switch` style.

File: iothub/device/src/Transport/AmqpIot/AmqpIotSendingLink.cs [hunk 13]
```diff
@@ -72,46 +73,44 @@ internal bool IsClosing()
...
+                Logging.Enter(this, nameof(SendMessagesAsync));
             }
 
             // List to hold messages in AMQP friendly format
             var messageList = new List<Data>();
 
             foreach (Message message in messages)
             {
-                using (AmqpMessage amqpMessage = AmqpIoTMessageConverter.MessageToAmqpMessage(message))
+                using AmqpMessage amqpMessage = AmqpIotMessageConverter.MessageToAmqpMessage(message);
```
Comment (drwill-ms) [inline iothub/device/src/Transport/AmqpIot/AmqpIotSendingLink.cs:108]: New using style

File: iothub/device/tests/AuthenticationMethod/ModuleAuthenticationWithTokenRefreshTests.cs [hunk 14]
```diff
@@ -1,12 +1,14 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
```
Comment (barustum) [inline iothub/device/tests/AuthenticationMethod/ModuleAuthenticationWithTokenRefreshTests.cs:2]: duplicate file header

File: iothub/device/src/Transport/AmqpIot/AmqpIotSession.cs [hunk 15]
```diff
@@ -297,53 +289,53 @@ TimeSpan timeout
...
             {
                 LinkName = linkSuffix,
                 Role = true,
                 TotalLinkCredit = prefetchCount,
                 AutoSendFlow = prefetchCount > 0,
-                Source = new Source() { Address = BuildLinkAddress(deviceIdentity, deviceTemplate, moduleTemplate) },
-                Target = new Target() { Address = deviceIdentity.IotHubConnectionString.DeviceId }
+                Source = new Source { Address = BuildLinkAddress(deviceIdentity, deviceTemplate, moduleTemplate) },
+                Target = new Target { Address = deviceIdentity.IotHubConnectionString.DeviceId },
+                SndSettleMode = senderSettleMode,
```
Comment (drwill-ms) [inline iothub/device/src/Transport/AmqpIot/AmqpIotSession.cs:305]: These properties should be able to be in the initializer list

File: iothub/device/tests/AuthenticationMethod/DeviceAuthenticationWithTokenRefreshTests.cs [hunk 16]
```diff
@@ -1,12 +1,14 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
```
Comment (barustum) [inline iothub/device/tests/AuthenticationMethod/DeviceAuthenticationWithTokenRefreshTests.cs:2]: duplicate

File: iothub/device/src/Transport/AmqpIot/AmqpIotTransportLog.cs [hunk 17]
```diff
@@ -23,7 +23,7 @@ public override void AmqpAddSession(object source, object session, ushort localC
         {
             if (Logging.IsEnabled)
             {
-                Logging.Associate(source, session, "AmqpAddSession:", $"{" source:"}{source}.{" session:"}{session}.{" localChannel:"}.{localChannel}.{" remoteChannel:"}.{remoteChannel}");
```
Comment (drwill-ms) [inline iothub/device/src/Transport/AmqpIot/AmqpIotTransportLog.cs:26]: I just... can't...

We string interpolate and include some string literals, but wrap them in curly braces to format in?

**Review by abhipsaMisra — APPROVED** [review]
Only comment is about the variable names still having the upper case T, but I am not sure if you intended to update those as a part of this PR.
Otherwise, looks good!

File: iothub/device/src/Transport/AmqpIot/AmqpIotLinkFactory.cs (hunk 7 above)
Comment (drwill-ms) [reply iothub/device/src/Transport/AmqpIot/AmqpIotLinkFactory.cs:40]: Good question. Removing it causes this error.

> error CS8400: Feature 'target-typed conditional expression' is not available in C# 8.0. Please use language version 9.0 or greater.

Either can have the cast to make the issue go away. If both have it, the IDE thinks one is unnecessary.

Weird, right?

**Review by barustum — APPROVED** [review]

**Review by azabbasi — APPROVED** [review]

**Pull request merged**